PHL-Patch-Reason: implicit-nullable-param
PHL-Patch-Kind: rewrite
PHL-Patch-Note: 9 parameters in php-console/php-console spell php 8.4's deprecated `T $x = null`; each gets the explicit `?T`. Generated by tests/vendor/tools/fix-implicit-nullable.php. Split from the doctrine/couchdb patch on purpose -- one patch per PACKAGE, so `composer reinstall` of one does not leave a patch half-applied.

diff -ruN a/vendor/php-console/php-console/src/PhpConsole/Auth.php b/vendor/php-console/php-console/src/PhpConsole/Auth.php
--- a/vendor/php-console/php-console/src/PhpConsole/Auth.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/src/PhpConsole/Auth.php	2026-08-25 23:42:24.410702186 -0300
@@ -48,7 +48,7 @@
 	 * @param ClientAuth|null $clientAuth
 	 * @return ServerAuthStatus
 	 */
-	public final function getServerAuthStatus(ClientAuth $clientAuth = null) {
+	public final function getServerAuthStatus(?ClientAuth $clientAuth = null) {
 		$serverAuthStatus = new ServerAuthStatus();
 		$serverAuthStatus->publicKey = $this->getPublicKey();
 		$serverAuthStatus->isSuccess = $clientAuth && $this->isValidAuth($clientAuth);
diff -ruN a/vendor/php-console/php-console/src/PhpConsole/Helper.php b/vendor/php-console/php-console/src/PhpConsole/Helper.php
--- a/vendor/php-console/php-console/src/PhpConsole/Helper.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/src/PhpConsole/Helper.php	2026-08-25 23:42:24.416841099 -0300
@@ -37,7 +37,7 @@
 		 * @throws \Exception
 		 * @return Connector
 		 */
-		public static function register(Connector $connector = null, Handler $handler = null) {
+		public static function register(?Connector $connector = null, ?Handler $handler = null) {
 			if(static::$connector) {
 				throw new \Exception('Helper already registered');
 			}
diff -ruN a/vendor/php-console/php-console/src/PhpConsole/OldVersionAdapter.php b/vendor/php-console/php-console/src/PhpConsole/OldVersionAdapter.php
--- a/vendor/php-console/php-console/src/PhpConsole/OldVersionAdapter.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/src/PhpConsole/OldVersionAdapter.php	2026-08-25 23:42:24.417235457 -0300
@@ -48,7 +48,7 @@
 		 * @throws \Exception
 		 * @return Connector
 		 */
-		public static function register(Connector $connector = null, Handler $handler = null) {
+		public static function register(?Connector $connector = null, ?Handler $handler = null) {
 		}
 
 		/**
diff -ruN a/vendor/php-console/php-console/src/PhpConsole/PsrLogger.php b/vendor/php-console/php-console/src/PhpConsole/PsrLogger.php
--- a/vendor/php-console/php-console/src/PhpConsole/PsrLogger.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/src/PhpConsole/PsrLogger.php	2026-08-25 23:42:24.417620755 -0300
@@ -44,7 +44,7 @@
 	 * @param Dumper|null $contextDumper
 	 * @param int|array $ignoreTraceCalls Ignore tracing classes by name prefix `array('PhpConsole')` or fixed number of calls to ignore
 	 */
-	public function __construct(Connector $connector = null, Dumper $contextDumper = null, $ignoreTraceCalls = 1) {
+	public function __construct(?Connector $connector = null, ?Dumper $contextDumper = null, $ignoreTraceCalls = 1) {
 		$this->connector = $connector ?: Connector::getInstance();
 		$this->contextDumper = $contextDumper ?: $this->connector->getDumper();
 		$this->ignoreTraceCalls = $ignoreTraceCalls;
diff -ruN a/vendor/php-console/php-console/test/Test/Remote/EvaluateTest.php b/vendor/php-console/php-console/test/Test/Remote/EvaluateTest.php
--- a/vendor/php-console/php-console/test/Test/Remote/EvaluateTest.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/test/Test/Remote/EvaluateTest.php	2026-08-25 23:42:24.428047013 -0300
@@ -18,7 +18,7 @@
 		$this->request->addScript('init_default_handler');
 	}
 
-	protected function setRequestEval($code, $signature = null, Auth $auth = null) {
+	protected function setRequestEval($code, $signature = null, ?Auth $auth = null) {
 		if(!$auth) {
 			$auth = $this->setRequestAuth();
 		}
diff -ruN a/vendor/php-console/php-console/test/Test/Remote/Test.php b/vendor/php-console/php-console/test/Test/Remote/Test.php
--- a/vendor/php-console/php-console/test/Test/Remote/Test.php	2019-07-25 00:43:28.000000000 -0300
+++ b/vendor/php-console/php-console/test/Test/Remote/Test.php	2026-08-25 23:42:24.429731207 -0300
@@ -167,7 +167,7 @@
 		return $auth;
 	}
 
-	protected function setConnectorAuth($password = null, $publicKeyByIp = true, $clientIp = '127.0.0.1', Request $request = null) {
+	protected function setConnectorAuth($password = null, $publicKeyByIp = true, $clientIp = '127.0.0.1', ?Request $request = null) {
 		if($password === null) {
 			$password = static::getClientEmulator()->getSecretKey();
 		}
